Taskfile.yml

```diff
@@ -209,56 +215,93 @@ tasks:
     vars:
       CHECKSUM_FILE: "{{.G_BUILD_DIR}}/{{.TASK}}.md5"
       OUTPUT_DIR: "{{.G_WEBUI_NODEJS_BUILD_DIR}}"
+    cmds:
+      - task: "nodejs"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          NODEJS_VERSION: "v14.21.3"
+          OUTPUT_DIR: "{{.OUTPUT_DIR}}"
+
+  core-submodules:
+    internal: true
+    dir: "components/core"
+    vars:
+      CHECKSUM_FILE: "{{.G_BUILD_DIR}}/{{.TASK}}.md5"
+      OUTPUT_DIR: "submodules"
     deps:
       - "init"
       - task: "validate-checksum"
         vars:
           CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
           DATA_DIR: "{{.OUTPUT_DIR}}"
     cmds:
-      - task: "nodejs"
-        vars:
-          NODEJS_VERSION: "v14.21.3"
-          OUTPUT_DIR: "{{.OUTPUT_DIR}}"
+      - "tools/scripts/deps-download/download-all.sh"
       # This command must be last
       - task: "compute-checksum"
         vars:
           DATA_DIR: "{{.OUTPUT_DIR}}"
           OUTPUT_FILE: "{{.CHECKSUM_FILE}}"
     sources:
       - "{{.TASKFILE}}"
+      - ".gitmodules"
+      - "tools/scripts/deps-download/**/*"
     generates: ["{{.CHECKSUM_FILE}}"]
 
-  core-submodules:
+  download-tar-and-extract:
     internal: true
-    dir: "components/core"
-    vars:
-      CHECKSUM_FILE: "{{.G_BUILD_DIR}}/{{.TASK}}.md5"
-      OUTPUT_DIR: "submodules"
+    requires:
+      vars: ["CHECKSUM_FILE", "FILE_BASE_DIR", "FILE_TAR_NAME", "OUTPUT_DIR", "URL_PREFIX"]
+    label: "Download & Extract {{.FILE_TAR_NAME}}"
     deps:
       - "init"
       - task: "validate-checksum"
         vars:
           CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
           DATA_DIR: "{{.OUTPUT_DIR}}"
     cmds:
-      - "tools/scripts/deps-download/download-all.sh"
+      - "rm -rf '{{.OUTPUT_DIR}}'"
+      - "mkdir -p '{{.OUTPUT_DIR}}'"
+      - >-
+        curl --fail --location --show-error
+        "{{.URL_PREFIX}}/{{.FILE_TAR_NAME}}"
+        -o "{{.OUTPUT_DIR}}/{{.FILE_TAR_NAME}}"
+      - "tar xf '{{.OUTPUT_DIR}}/{{.FILE_TAR_NAME}}' --directory '{{.OUTPUT_DIR}}'"
+      - >-
+        mv "{{.FILE_BASE_DIR}}/"*
+        "{{.OUTPUT_DIR}}"
+      - "rm -rf '{{.FILE_BASE_DIR}}'"
+      - "rm -f '{{.OUTPUT_DIR}}/{{.FILE_TAR_NAME}}'"
       # This command must be last
       - task: "compute-checksum"
         vars:
           DATA_DIR: "{{.OUTPUT_DIR}}"
           OUTPUT_FILE: "{{.CHECKSUM_FILE}}"
     sources:
-      - "{{.TASKFILE}}"
-      - ".gitmodules"
-      - "tools/scripts/deps-download/**/*"
+      - "{{.TASKFILE_DIR}}/Taskfile.yml"
     generates: ["{{.CHECKSUM_FILE}}"]
 
+  meteor:
```

Comment:
Should we make this an internal task?

Reply:
I didn't mark this as internal because this Meteor.js installation can be used to launch the WebUI manually in development mode, without developers manually installing Meteor.js in their environment. 

With that said, these days it is hard to launch the WebUI without its dependencies in the CLP package. If developers do not have a CLP package on hand and have to build the package, this task would have been executed by the time the package is built.

Nonetheless we probably want to make `webui-nodejs` as internal, since that task is only used for initializing the meteor-built webui.